### ISSUE TITLE
Converting the Rust logo doesn't work

### ISSUE THREAD

**Issue by msrd0** [typst/svg2pdf#17]
I tried converting the Rust logo (https://upload.wikimedia.org/wikipedia/commons/d/d5/Rust_programming_language_black_logo.svg) using this tool but it looks a little off:

![image](https://user-images.githubusercontent.com/7853372/185191716-3484724b-cd9d-412d-ae47-b3db90d9f24e.png)

**Comment by LaurenzV** [comment]
I tried debugging this, and I think the reason is that similarly to #9, the transformation of a group doesn't seem to apply to masks. Here is a mininmal examples that reproduces this:

```rust
<svg version="1.1" height="200" width="200" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <g id="logo" transform="translate(50, 50)">
        <g id="gear" mask="url(#holes)">
            <rect width="100" height="100" fill="green" />
        </g>
        <mask id="holes" >
            <rect width="30" height="30" fill="white"/>
        </mask>
    </g>
</svg>
```

I tried finding the issue in the code and I suspect that something is wrong in the `apply_mask` method, but unfortunately I don't really understand what is going on there with the calculations, so I haven't been able to fix this.

**Comment by reknih** [comment]
I'm sorry about having closed the issue, the text in PR #19 triggered the auto-close.
